main.py

```diff
@@ -0,0 +1,258 @@
+"""
+This main python file contains the backend logic of the website, including
+the reception of Cloud Pub/Sub messages and communication with GCS and datastore.
+"""
+
+# Import all necessary libraries.
+import webapp2
+import jinja2
+import os
+import logging
+import json
+import urllib
+import collections
+import cloudstorage as gcs
+import googleapiclient.discovery
+from google.appengine.ext import ndb
+from google.appengine.ext import blobstore
+from google.appengine.api import images
+
+# Set up jinja2 for HTML templating.
+template_dir = os.path.join(os.path.dirname(__file__), 'templates')
+jinja_environment = jinja2.Environment(loader = jinja2.FileSystemLoader(template_dir))
+
+# Constants. Note: Change THUMBNAIL_BUCKET and PHOTO_BUCKET to
+# be applicable to your project.
+THUMBNAIL_BUCKET = 'thumbnails-bucket'
```

Comment:
Constants should be defined right after imports, so move these above the jinja2 initialization.